### PR TITLE
[Merged by Bors] - feat(GroupTheory/Subgroup/Basic): ker_{fst,snd}

### PR DESCRIPTION
**Pull request by pechersky**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/Subgroup/Basic.lean [hunk 1]
```diff
@@ -2887,6 +2887,18 @@ instance (priority := 100) normal_ker (f : G →* M) : f.ker.Normal :=
 #align monoid_hom.normal_ker MonoidHom.normal_ker
 #align add_monoid_hom.normal_ker AddMonoidHom.normal_ker
 
+@[to_additive (attr := simp)]
+lemma ker_fst (G H : Type _) [Group G] [Group H] :
```
[the remark this answers is not in the document]
Comment (pechersky) [reply Mathlib/GroupTheory/Subgroup/Basic.lean:2891]: Would you like them to be explicit or implicit variables?

File: Mathlib/GroupTheory/Subgroup/Basic.lean [hunk 2]
```diff
@@ -2887,6 +2887,16 @@ instance (priority := 100) normal_ker (f : G →* M) : f.ker.Normal :=
 #align monoid_hom.normal_ker MonoidHom.normal_ker
 #align add_monoid_hom.normal_ker AddMonoidHom.normal_ker
 
+@[to_additive (attr := simp)]
+lemma ker_fst : ker (fst G G') = (.prod ⊥ ⊤) := by
+  ext
+  simp [mem_ker, Subgroup.mem_prod]
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/Subgroup/Basic.lean:2893]:
```suggestion
lemma ker_fst : ker (fst G G') = (.prod ⊥ ⊤) := ext fun _ => (and_true_iff _).symm
```

File: Mathlib/GroupTheory/Subgroup/Basic.lean [hunk 3]
```diff
@@ -2887,6 +2887,16 @@ instance (priority := 100) normal_ker (f : G →* M) : f.ker.Normal :=
...
+
+@[to_additive (attr := simp)]
+lemma ker_snd : ker (snd G G') = (.prod ⊤ ⊥) := by
+  ext
+  simp [mem_ker, Subgroup.mem_prod]
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/Subgroup/Basic.lean:2898]:
```suggestion
lemma ker_snd : ker (snd G G') = (.prod ⊤ ⊥) := ext fun _ => (true_and_iff _).symm
```

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks! Some golfs to consider

File: Mathlib/GroupTheory/Subgroup/Basic.lean [hunk 4]
```diff
@@ -2887,6 +2887,12 @@ instance (priority := 100) normal_ker (f : G →* M) : f.ker.Normal :=
 #align monoid_hom.normal_ker MonoidHom.normal_ker
 #align add_monoid_hom.normal_ker AddMonoidHom.normal_ker
 
+@[to_additive (attr := simp)]
+lemma ker_fst : ker (fst G G') = (.prod ⊥ ⊤) := ext fun _ => (and_true_iff _).symm
```
Comment (eric-wieser) [inline Mathlib/GroupTheory/Subgroup/Basic.lean:2891]: Whoops, I guess this wasn't the namespace I thought it was 
```suggestion
lemma ker_fst : ker (fst G G') = (.prod ⊥ ⊤) := SetLike.ext fun _ => (and_true_iff _).symm
```

**Pull request closed without merging**